benchmarks/serde-benchmarks/build.gradle.kts

```diff
@@ -0,0 +1,129 @@
+            val resourcesPrefix = if (name.endsWith("Test")) "test-" else  ""
+            // the name is always the platform followed by a suffix of either "Main" or "Test" (e.g. jvmMain, commonTest, etc)
+            val platform = name.substring(0, name.length - 4)
+            kotlin.srcDir("$platform/$srcDir")
+            resources.srcDir("$platform/${resourcesPrefix}resources")
+            languageSettings.progressiveMode = true
+            experimentalAnnotations.forEach { languageSettings.optIn(it) }
+        }
+
+        val kotlinxBenchmarkVersion: String by project
+        val coroutinesVersion: String by project
+        commonMain {
+            dependencies {
+                implementation("org.jetbrains.kotlinx:kotlinx-benchmark-runtime:$kotlinxBenchmarkVersion")
+                implementation(project(":runtime:serde:serde-json"))
+                implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutinesVersion")
+            }
+        }
+        val jvmMain by getting {
+            kotlin.srcDir("${project.buildDir}/generated-src/src")
+        }
+    }
+}
+
+benchmark {
+    targets {
+        register("jvm")
+    }
+
+    configurations {
+        getByName("main") {
+            iterations = 5
+            iterationTime = 1
+            iterationTimeUnit = "s"
+            warmups = 7
+            outputTimeUnit = "ms"
+        }
+    }
+}
+
+// Workaround for https://github.com/Kotlin/kotlinx-benchmark/issues/39
+afterEvaluate {
+    tasks.named<org.gradle.jvm.tasks.Jar>("jvmBenchmarkJar") {
+        duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+    }
+}
+
+tasks["smithyBuildJar"].enabled = false
+
+val codegen by configurations.creating
+
+dependencies {
+    codegen(project(":benchmarks:serde-benchmarks-codegen"))
+}
+
+val generateSdk = tasks.create<SmithyBuild>("generateSdk") {
+    group = "codegen"
+    classpath = configurations.getByName("codegen")
+    println(configurations.getByName("codegen"))
+    inputs.file(projectDir.resolve("smithy-build.json"))
+    inputs.files(configurations.getByName("codegen"))
+}
+
+data class BenchmarkModel(val name: String) {
+    val projectionRootDir: File
+        get() = project.file("${project.buildDir}/smithyprojections/${project.name}/$name/kotlin-codegen/src/main/kotlin").absoluteFile
+
+    val sourceSetRootDir: File
+        get() = project.file("${project.buildDir}/generated-src/src").absoluteFile
+}
+
+val benchmarkModels = listOf("twitter").map{ BenchmarkModel(it) }
+
+
+val stageGeneratedSources = tasks.register("stageGeneratedSources") {
+    group = "codegen"
+    dependsOn(generateSdk)
+    doLast {
+        benchmarkModels.forEach {
+            copy {
+                from("${it.projectionRootDir}")
+                into("${it.sourceSetRootDir}")
+                include("**/model/*.kt")
+                include("**/transform/*.kt")
+                exclude("**/transform/*OperationSerializer.kt")
```

Comment:
(curious) question

Is the reason for exclusion here that these generated files are invalid due to implementation details of the benchmark protocol, or just cleanliness?  (if you remove these exclusions does the benchmarking still work?)

Reply:
Sort of multifaceted. (1) The benchmark protocol generator doesn't generate anything other than the "document" serializers/deserializers and (2) the benchmark test dependencies don't include any of the other modules other than serde. We would have to add dependencies on e.g. `http`, etc for these to compile.